src/main/java/de/qabel/desktop/hockeyapp/VersionClient.java

```diff
@@ -0,0 +1,122 @@
+        String responseContent = EntityUtils.toString(response.getEntity());
+
+        parseVersionsResponse(responseContent);
+    }
+
+    void findAndLoadVersion(String shortVersion) throws IOException {
+        getVersions().forEach(version -> {
+            if (shortVersion.equals(version.getShortVersion())) {
+                setVersion(version);
+            }
+        });
+        if (version == null) {
+            createVersion(shortVersion);
+        }
+    }
+
+    void createVersion(String version) throws IOException, JSONException {
+        HttpPost request = config.getHttpPost(API_VERSIONS_NEW);
+
+        List<NameValuePair> parameters = buildCreateParameters(version);
+        request.setEntity(new UrlEncodedFormEntity(parameters, HTTP.UTF_8));
+
+        HttpResponse response = httpClient.execute(request);
+        String responseContent = EntityUtils.toString(response.getEntity());
+
+        parseVersionCreateResponse(responseContent);
+    }
+
+    List<HockeyAppVersion> parseVersionsResponse(String responseContent) throws IOException {
+        versions.clear();
+        try {
+            JSONObject parsedJson = new JSONObject(responseContent);
+            JSONArray versionArray = parsedJson.getJSONArray("app_versions");
+            for (int i = 0; i < versionArray.length(); i++) {
+                JSONObject jsonObj = versionArray.getJSONObject(i);
+                int versionId = jsonObj.getInt("version");
+                String shortVersion = jsonObj.getString("shortversion");
+                versions.add(new HockeyAppVersion(versionId, shortVersion));
+            }
+            return versions;
+        } catch (JSONException e) {
+            throw new IOException("returned JSON was invalid", e);
+        }
+    }
+
+    void parseVersionCreateResponse(String responseContent) throws IOException {
+        try {
+            JSONObject parsedJson = new JSONObject(responseContent);
+            int versionId = parsedJson.getInt("id");
+            String shortVersion = parsedJson.getString("shortversion");
+            setVersion(new HockeyAppVersion(versionId, shortVersion));
+        } catch (JSONException e) {
+            throw new IOException("returned JSON was invalid", e);
+        }
+    }
+
+    List<NameValuePair> buildCreateParameters(String version) {
+        List<NameValuePair> parameters = new ArrayList<>();
+        parameters.add(new BasicNameValuePair("bundle_short_version", version));
+        return parameters;
+    }
+
+
+    public HockeyAppVersion getVersion() throws IOException {
+        if (version == null) {
+            findAndLoadVersion(config.getAppVersion());
+        }
+        return version;
+    }
+
+    public void setVersion(HockeyAppVersion version) {
+        this.version = version;
+    }
+
+    List<HockeyAppVersion> getVersions() {
+        return versions;
+    }
+
+    public void setVersions(List<HockeyAppVersion> versions) {
```

Comment:
don't think this needs to be public. Remove the `public` to make it package local and use it in the tests. Otherwise it's part of the classes public API.